lib/active_model/serializer/fieldset.rb

```diff
@@ -20,11 +20,7 @@ def fields_for(type)
       private
 
       def parsed_fields
-        if raw_fields.is_a?(Hash)
-          raw_fields.each_with_object({}) { |(k, v), h| h[k.to_sym] = v.map(&:to_sym) }
-        else
-          {}
-        end
+        JSONAPI::IncludeDirective.new(raw_fields).to_hash
```

Comment:
well, the issue IS with parsed_fields because it removes the way you have to specify relationships.

```
root_type: [:relationship_type]
```